### ISSUE TITLE
Observation on Trigger groups and naming the levels

### ISSUE THREAD

**Issue by pbyrne-ms** [autoSteve/acMqtt#6]
Hi Steve,
Noticed a little issue, and admittedly I probably caused this as I asked for it originally :)

When discovering trigger groups, if levels are not actually named in the SHAC/NAC on the relevant trigger group, nothing is published,
I think because of this:

      GetCBusLevelTag(net, app, group, level)
      tag = GetCBusLevelTag((tNetAC(net)), 202, group, tonumber(i))
      --log("This Tag is : " .. tag)
      if tag then  -- Will not pass this test if the levels are not named in SHAC...so names have to be declared..
        					 --  
        log("Tag is : " .. tag)
        boid = oid..'_'..i
	-- and so on....	

Even when we declare the levels required by adding the keyword tag 'lvl=0-63-127-199-255', if those numbers don't really have names assigned to their respective level number, it does not publish them, and the current logging did not really reveal the reason..

Not really sure of the best solution... a simple fix is to just modify the README with a note to state all required levels have names defined..
Perhaps there is a more robust technical solution?

Personally, I do normally name all my levels, but I stumbled across the 1 out of 27 where I didn't today!

-P

**Comment by autoSteve** [comment]
This is intentional behaviour I built in, but you're right to point out that silence is not always golden. Consider a scenario where you do not specify any levels in the lvl= keyword. The script would attempt to publish all levels, only publishing named ones.

My first attempt did not allow lvl= selection, so silence was better. Continued silence is because I did not consider it further.

Publish it as "Level n"? With no lvl= keyword there would be 256 items published, most of which would be useless.

Log an error message? With no lvl= keyword there would be a tonne of logged errors (well, up to 256 of them). I suppose a single "No level names" log entry would modify that behaviour, but this would not pick up that a functional level desired actually has no name.

In the case where there _is_ a lvl= keyword a logged error or publish as "Level n" makes more sense (either would expose the issue). Maybe a combination of "No level names" for no lvl= keyword (which would mask an underlying issue), or for lvl= specific errors or generic publishing (maybe even with behaviour modified using a variable in the code).

Preference?
